test/rateLimiters/slidingWindowCounter.test.ts

```diff
@@ -0,0 +1,476 @@
+                    (await limiter.processRequest(user4, timestamp + WINDOW_SIZE * 1.25, 5)).tokens
+                ).toBe(10);
+
+                // currentTokens (in current fixed window): 0
+                // previousTokens (in previous fixed window): 8
+                const count1 = await getWindowFromClient(client, user4);
+                expect(count1.currentTokens).toBe(0);
+                expect(count1.previousTokens).toBe(initRequest);
+            });
+        });
+
+        test('rolling window at 50% blocks requests over allowed limit set by formula', async () => {
+            // 50% of rolling window present in previous fixed window
+            // 1.5*60000 = 90000 (time after initial fixedWindowStart
+            // to set rolling window at 50% of previous fixed window)
+
+            // to set initial fixedWindowStart
+            await setTokenCountInClient(client, user4, 0, null, timestamp);
+
+            const initRequest = 8;
+
+            // large request at very end of first fixed window
+            await limiter.processRequest(user4, timestamp + WINDOW_SIZE, initRequest);
+
+            // 7 + 8 * .5 = 11, over capacity (request should be blocked)
+            expect(
+                (await limiter.processRequest(user4, timestamp + WINDOW_SIZE * 1.5, 7)).tokens
+            ).toBe(10);
+
+            // currentTokens (in current fixed window): 0
+            // previousTokens (in previous fixed window): 8
+            const count = await getWindowFromClient(client, user4);
+            expect(count.currentTokens).toBe(0);
+            expect(count.previousTokens).toBe(initRequest);
+        });
+
+        test('rolling window at 25% blocks requests over allowed limit set by formula', async () => {
+            // 25% of rolling window present in previous fixed window
+            // 1.75*60000 = 105000 (time after initial fixedWindowStart
+            // to set rolling window at 25% of previous fixed window)
+
+            // to set initial fixedWindowStart
+            await setTokenCountInClient(client, user4, 0, null, timestamp);
+
+            const initRequest = 8;
+
+            // large request at very end of first fixed window
+            await limiter.processRequest(user4, timestamp + WINDOW_SIZE, initRequest);
+
+            // 9 + 8 * .25 = 11, over capacity (request should be blocked)
+            expect(
+                (await limiter.processRequest(user4, timestamp + WINDOW_SIZE * 1.75, 9)).tokens
+            ).toBe(10);
+
+            // currentTokens (in current fixed window): 0
+            // previousTokens (in previous fixed window): 8
+            const count = await getWindowFromClient(client, user4);
+            expect(count.currentTokens).toBe(4);
+            expect(count.previousTokens).toBe(initRequest);
+        });
+    });
+
+    describe('SlidingWindowCounter functions as expected', () => {
+        afterEach(() => {
+            client.flushall();
+        });
+
```

Comment:
Add a test case for a request that comes in at `timestamp _ 2 * WINDOW_SIZE` (e.g. not in the current or next window).